File: code/1/solution.py
class Solution:
    def twoSum(self, nums: list, target: int) -> list:
        sort_num = sorted(nums)
        i = 0;
        j = len(nums) - 1
        while True:
            if i >= j :
                return []
            if sort_num[i] + sort_num[j] == target :
                if sort_num[j] == sort_num[i] :
                    a = nums.index(sort_num[i])
                    b = nums.index(sort_num[j], a + 1)
                else :
                    a = nums.index(sort_num[i])
                    b = nums.index(sort_num[j])
                return [a,b]
            if sort_num[i] + sort_num[j] > target :
                j -= 1
                continue
            if sort_num[i] + sort_num[j] < target :
                i += 1
                continue
           

a = Solution()
b = a.twoSum([-1,-2,-3,-4,-5],-8)
print(b)

File: code/1/solution.php
<?php

class Solution {
    /**
     * 暴力法
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        for($i=0;i<count($nums) - 1;$i++){
            for($j=$i+1;$j<count($nums);$j++){
                if($nums[$i] + $nums[$j] == $target){
                        return [$i,$j];
                    }
                }
            }
            return [];
        }
    }

    /**
     * hashMap
     */
    function twoSum2($nums, $target) {
        for ($i=0;$i<count($nums);$i++){
            $ele = $target - $nums[$i];
            $exists = array_keys($nums, $ele);
            foreach($exists as $key){
                if($key != $i) return [$i, $key];
            }
        }
        return [];
    }

$a = new Solution();
$b = $a->twoSum2([1,2,7],9);
var_dump($b);